frontend/src/components/organisms/ProfileForm.tsx

```diff
@@ -39,8 +41,76 @@ interface ProfileFormProps {
 
 const ProfileForm = ({ userDetails }: ProfileFormProps) => {
   const router = useRouter();
+  const [updatePassword, updating, error] = useUpdatePassword(auth);
+  const [success, setSuccess] = React.useState<boolean>(false);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = React.useState<string>("");
 
-  // Future testing: What if user doesn't have a nickname, does the code break?
+  const handleErrors = (errors: any) => {
+    const errorParsed = errors?.split("/")[1].slice(0, -2);
+    switch (errorParsed) {
```

Comment:
Can we put these handleError functions in utils?

Reply:
There does seem to be a lot of duplication with the `handleError` functions; we probably should be able to abstract it into a helper method since the Firebase error codes are all standardized. Will look into this

Reply:
I agree but will take some care to duplicate it because the different functionalities have slightly different error codes/messages